manager/state/watch/watch.go

```diff
@@ -46,3 +46,8 @@ func (q *Queue) CallbackWatch(matcher events.Matcher) (eventq chan events.Event,
 func (q *Queue) Publish(item events.Event) {
 	q.broadcast.Write(item)
 }
+
+// Close closes the queue and frees the associated resources.
+func (q *Queue) Close() error {
+	return q.broadcast.Close()
```

Comment:
I think this has to go through and close each sink before closing the broadcast portion.